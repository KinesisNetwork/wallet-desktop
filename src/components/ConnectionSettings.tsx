import * as React from 'react'

import { ConnectionForm } from '@containers/ConnectionForm'
import { ConnectionSelector } from '@containers/ConnectionSelector'

export const ConnectionSettings: React.SFC = () => (
  <div className='vertical-spaced has-text-centered'>
    <h1 className='title-heading'>Manage Connections</h1>
    <div className='columns'>
      <div className='column'>
        <ConnectionSelector />
      </div>
      <div className='column'>
        <ConnectionForm />
      </div>
    </div>
  </div>
)

ConnectionSettings.displayName = 'ConnectionSettings'
