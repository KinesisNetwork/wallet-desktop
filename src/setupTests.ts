import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
