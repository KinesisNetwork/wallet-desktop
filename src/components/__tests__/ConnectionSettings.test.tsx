import { shallow } from 'enzyme'
import * as React from 'react'

import { ConnectionSettings } from '@components/ConnectionSettings'
import '../../setupTests'

describe('ConnectionSettings', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ConnectionSettings />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render a primary heading, a ConnectionSelector and a ConnectionForm', () => {
    const wrapper = shallow(<ConnectionSettings />)

    expect(wrapper.find('h1')).toHaveLength(1)
    expect(wrapper.find('h1').text()).toEqual('Manage Connections')
    expect(wrapper.find('Connect(ConnectionSelectorComponent)')).toHaveLength(1)
  })
})
