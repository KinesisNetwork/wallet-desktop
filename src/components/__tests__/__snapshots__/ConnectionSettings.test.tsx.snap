// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionSettings renders correctly 1`] = `
<div
  className="vertical-spaced has-text-centered"
>
  <h1
    className="title-heading"
  >
    Manage Connections
  </h1>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <Connect(ConnectionSelector) />
    </div>
    <div
      className="column"
    >
      <Connect(ConnectionForm) />
    </div>
  </div>
</div>
`;
